EpikCord/__init__.py

```diff
@@ -1977,6 +1985,106 @@ def unload_section(self, section: Union[CommandsSection, EventsSection]):
 #     def __init__(self, client: Client,data: dict):
 #         super().__init__(data)
 
+class VoiceWebsocketClient:
+    #This class shouldnt be used for 
+    def __init__(self, client:Client, **kwargs):
+        self.client = client
+        self.guild_id = kwargs.get("guild_id")
+        self.channel_id = kwargs.get("channel_id")
+        self.HELLO = 8
+        self.IDENTIFY = 0
+        self.HEARTBEAT = 3
+        self.READY = 2
+        self.voice_connected = False
+        self.server_set = False
+        self.state_set = False
+        self.sequence = None
+
+
+        self.client.events["voice_state_update"] = self.voice_state_update
+        self.client.events["voice_server_update"] = self.voice_server_update            
+    async def voice_state_update(self,data:dict):
+        if self.voice_connected == True:
+            if self.state_set:
+                logger.info("Ignoring extra voice event")
+            self.session_id = data["session_id"]
+            self.state_set = True
+    async def voice_server_update(self,data:dict):
+        if self.voice_connected == True:
+            if self.server_set:
+                logger.info("Ignoring extra voice event")
+                return
+            voice_data = data["d"]
+            self.token = voice_data["token"]
+            self.endpoint = voice_data["endpoint"]
+            self.server_set = True
+
+    async def connect(self, guild_id:Optional[str] = None, channel_id:Optional[str] = None, **kwargs):
+        self.guild_id = guild_id if guild_id else self.guild_id
+        self.channel_id = channel_id if channel_id else self.channel_id
+        await self.client.send_json({
+            "op": 4,
+            "d" : {
+                "guild_id": self.guild_id,
+                "channel_id": self.channel_id,
+                "self_mute": kwargs.get("self_mute", False),
+                "self_deaf" : kwargs.get("self_deaf", False)
+            }
+        })
+        self.voice_connected = True 
+        self.ws = await self.client.http.ws_connect(f"wss://{self.endpoint}/?v=4")
+        while not self.ws.closed:
+            await self.handle_ws_events()
+        await self.identify()
+        
+        await asyncio.sleep(1) # wait for some while for getting events       
+        async def heartbeat():
+            while True:
+                await self.heartbeat(False)
+        asyncio.create_task(heartbeat())
+
+    async def send_json(self, json):
+        '''Sends JSON data to the ``voice`` websocket connection
+        Note: Fails when used before ``connect()`` **and** after client disconnects'''
+        await self.ws.send_json(json)
+        logger.info(f"Sent {json} to Discord.")
+
+    async def handle_ws_events(self):
```

Comment:
Rename it `handle_events` for consistency. Even if it's different from `WebsocketClient`, it will overwrite any pre-existing.